src/App.js

```diff
@@ -0,0 +1,138 @@
+import {ethers} from "ethers"
+import {useEffect, useState} from "react";
+import "./App.css";
+import NFTFactory from "./artifacts/contracts/NFTFactory.sol/NFTFactory.json";
+import NFTFactoryWithAddress from "./NFTFactory.deployed.json";
+
+
+function App() {
+  const [contractData, setContractData] = useState({
+      owner: '',
+      contract: {},
+      cost: '',
+      balance: '',
+  });
+  const [inputValue, setInputValue] = useState("");
+  const [senderAddress, setSenderAddress] = useState("");
+  const [error, setError] = useState("");
+
+  const {contract, owner, cost, balance} = contractData;
+
+  const mint = async (isForOwner) => {
+    if (inputValue) {
+      const options = {
+        value: ethers.utils.parseEther(inputValue),
+        // from: owner //@TODO when need to change owner msg.from address
+      };
+      try {
+        const transaction = await contract.mint(isForOwner ? owner : senderAddress, options);
+
+        await transaction.wait();
+        setError('');
+
+        await initialConnection();
+      } catch (e) {
+        setError(e.data !== undefined ? e.data.message : e.message);
+      }
+    } else {
+      alert("Fill input value")
+    }
+  }
+
+  const initialConnection = async () => {
+    if (typeof window.ethereum !== "undefined") {
+      const data = await window.ethereum.request({method: "eth_requestAccounts"});
+      const provider = new ethers.providers.Web3Provider(window.ethereum);
+      const singer = provider.getSigner();
+      const network = await provider.getNetwork();
+
+      let contract;
+      if (network.name === "rinkeby") {
+        contract = new ethers.Contract(
+          NFTFactoryWithAddress.address,
+          NFTFactoryWithAddress.abi,
+          singer
+        );
+      } else {
+        contract = new ethers.Contract(
+          NFTFactoryWithAddress.address,
+          NFTFactory.abi,
+          singer
+        );
+      }
+
+      const costHexadecimal = await contract.cost();
+      const balance = await provider.getBalance(data[0]);
+
+      setContractData({
+        contract,
+        owner: data[0],
+        cost: ethers.utils.formatEther(costHexadecimal),
+        balance: ethers.utils.formatEther(balance),
+      });
+    } else {
+      console.log("ssadasdas");
```

Comment:
you can have a user friendly alert here, like: "Your browser does not support metamask. Please use other browser"... :)))